blueocean-core-js/src/js/services/DefaultSSEHandler.js

```diff
@@ -66,12 +66,18 @@ export class DefaultSSEHandler {
 
         for (const key of this.branchPagerKeys(event)) {
             const pager = this.pagerService.getPager({ key });
-            this.activityService.fetchActivity(runSelf, { overrideQueuedState }).then(d => {
-                if (pager && !pager.has(runSelf)) {
-                    pager.insert(runSelf);
-                }
-                this.pipelineService.updateLatestRun(d);
-            });
+            // If we don't have a pager, then the app in the browser has never visited that
+            // Job, which means there's no need to fetch the data associated with it because
+            // there's nothing in the app that needs to be kept up-to-date. The data will
+            // be fetched when the user visits that job/pipeline.
+            if (pager) {
+                this.activityService.fetchActivity(runSelf, { overrideQueuedState }).then(d => {
+                    if (!pager.has(runSelf)) {
+                        pager.insert(runSelf);
+                    }
+                    this.pipelineService.updateLatestRun(d);
```

Comment:
@kzantow we were thinking that this change (this whole PR) may not be warranted just yet, as it may have other subtle bugs introduced (ATH picked up just one). So unless we/you are sure - do you think it should be shelved? it is a valid improvement, but not a critical thing IMO (do you agree?)